kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java

```diff
@@ -82,6 +82,7 @@ public static void generalSetup() {
     public void setup() {
         unit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
         deploymentService.deploy(unit);
+        countDownListener.setSaveContent(false);
```

Comment:
Looks good to me, though, not really sure why this bean is being shared among the test cases.

Another different approach I've tried is to use the following inside `AfterClass` or even `After` annotated methods in both test cases:

- https://github.com/kiegroup/droolsjbpm-integration/blob/21879389f74c1c8ea44ae0269262db9c4e58990a/kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java
- https://github.com/kiegroup/droolsjbpm-integration/blob/21879389f74c1c8ea44ae0269262db9c4e58990a/kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/DeadlinesNotificationTest.java

`NotificationListenerManager.get().reset()` - By doing this, all notification listeners will be removed from the cache, and thus no one will be reused aftewards.

Reply:
Yes, good idea, that's a better approach. Thanks for suggestion @afalhambra !